db_addon/item_attributes_master.py

```diff
@@ -168,27 +205,70 @@ def export_db_addon_data():
+    valid_list_desc_str =    """        # NOTE: valid_list_description is automatically created by using item_attributes_master.py"""
+    valid_list_item_type =   """        # NOTE: valid_list_item_type is automatically created by using item_attributes_master.py"""
+    valid_list_calculation = """        # NOTE: valid_list_calculation is automatically created by using item_attributes_master.py"""
+
+    for db_addon_fct in ITEM_ATTRIBUTS[attribute]:
+        valid_list_str = f"""{valid_list_str}\n\
+          - {db_addon_fct!r:<40}"""
+
+        valid_list_desc_str = f"""{valid_list_desc_str}\n\
+          - '{ITEM_ATTRIBUTS[attribute][db_addon_fct]['description']:<}'"""
+
+        valid_list_item_type = f"""{valid_list_item_type}\n\
+          - '{ITEM_ATTRIBUTS[attribute][db_addon_fct]['item_type']:<}'"""
+
+        valid_list_calculation = f"""{valid_list_calculation}\n\
+          - '{ITEM_ATTRIBUTS[attribute][db_addon_fct]['calc']:<}'"""
+
+    valid_list_calculation = f"""{valid_list_calculation}\n\r"""
+
+    return valid_list_str, valid_list_desc_str, valid_list_item_type, valid_list_calculation
+
+def update_plugin_yaml_avm_data_type():
+    """Update ´'valid_list', 'valid_list_description', 'valid_list_item_type' and 'valid_list_calculation' of item attributes in plugin.yaml"""
+
+    yaml = ruamel.yaml.YAML()
+    yaml.indent(mapping=4, sequence=4, offset=4)
+    yaml.width = 200
+    yaml.allow_unicode = True
+    yaml.preserve_quotes = False
+
+    valid_list_str, valid_list_desc_str, valid_list_item_type_str, valid_list_calc_str = create_plugin_yaml_item_attribute_valids()
+
+    with open(FILENAME_PLUGIN, 'r', encoding="utf-8") as f:
+        data = yaml.load(f)
 
-        for title in ['description', 'item_type', 'calc']:
-            print(f'valid_list_{entry}:')
-            for func in ITEM_ATTRIBUTS[entry]:
-                print(f"    - '{ITEM_ATTRIBUTS[entry][func][title]}'")
-        print()
+    if data.get('item_attributes', {}).get(attribute):
+        data['item_attributes'][attribute]['valid_list'] = yaml.load(valid_list_str)
+        data['item_attributes'][attribute]['valid_list_description'] = yaml.load(valid_list_desc_str)
+        data['item_attributes'][attribute]['valid_list_item_type'] = yaml.load(valid_list_item_type_str)
+        data['item_attributes'][attribute]['valid_list_calculation'] = yaml.load(valid_list_calc_str)
 
+        with open(FILENAME_PLUGIN, 'w', encoding="utf-8") as f:
+            yaml.dump(data, f)
+        print(f"Successfully updated Attribut '{attribute}' in plugin.yaml!")
+    else:
+        print(f"Attribut '{attribute}' not defined in plugin.yaml")
 
 if __name__ == '__main__':
-    export_db_addon_data()
-    print()
-    print('--------------------------------------------------------------')
-    print()
-    export_for_plugin_yaml()
+    export_item_attributs_py()
+    for attribute in ITEM_ATTRIBUTS:
```

Comment:
E missing in attributes (also adjust all occurences)

Reply:
Corrected